gcd/src/test/java/org/spine3/server/storage/datastore/BigDataTester.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.storage.datastore;
+
+import com.google.common.base.Throwables;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.spine3.server.entity.EntityRecord;
+import org.spine3.server.entity.storage.EntityRecordWithColumns;
+import org.spine3.server.storage.RecordStorage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * A test utility that checks that the data read/write operations perform well (i.e. execute within
+ * the specified time
+ * limit) upon a big amount of records.
+ *
+ * <p>The default count of records is 500. Use {@link #setBulkSize(int)} to modify this number.
+ *
+ * <p>Note: the operations expected to be executed upon the whole bulk of records, not each record
+ * separately.
+ *
+ * <p>The test suits based on this utility are not expected to check if the read records match
+ * those that were written or not. Instead they should just check the time consumed by
+ * the operations and the consistency of the data (i.e. count of the records written and read).
+ *
+ * @author Dmytro Dashenkov
+ */
+public class BigDataTester<I> {
+
+    private static final int DEFAULT_BULK_SIZE = 500;
+
+    private final RecordStorage<I> storage;
+
+    private int bulkSize = DEFAULT_BULK_SIZE;
+    private EntrySupplier<I> entrySupplier;
+    private long writeMillisLimit;
+    private long readMillisLimit;
+
+    public BigDataTester(RecordStorage<I> target) {
+        this.storage = target;
+    }
+
+    public BigDataTester setBulkSize(int bulkSize) {
```

Comment:
Please have a `Builder` instead of a number of setters.

And for the `Builder`, please follow the conventions (checks for `null`s, default values, documentation etc.)